### PR TITLE
feat: [LW-9020] custom staking delegation

### PR DESCRIPTION
**Pull request by renanvalentin**

### CODE REVIEW

**Review by lucas-barros — COMMENTED** [review]
Why do we need typescript setup?

File: src/ui/app/components/transactionBuilder.jsx [hunk 1]
```diff
@@ -268,24 +341,76 @@ const TransactionBuilder = React.forwardRef(({ onConfirm }, ref) => {
...
+              </Link>
+              , connect your Nami wallet and delegate to a stake pool of your
+              choice
             </Text>
             <Box h="6" />
-            {data.error ? (
+            <Tooltip
+              label={`${data.pool.ticker} ${data.pool.name}`}
+              placement="top"
+              isOpen={data.pool.state === PoolStates.DONE ? undefined : false}
```
Comment (pczeglik-iohk) [inline src/ui/app/components/transactionBuilder.jsx:360]: when this is set to `true`?

File: src/ui/app/components/transactionBuilder.jsx [hunk 2]
```diff
@@ -268,24 +341,76 @@ const TransactionBuilder = React.forwardRef(({ onConfirm }, ref) => {
...
+                />
+                <InputRightElement {...poolRightElementStyle(data.pool)}>
+                  {data.pool.state === PoolStates.EDITING && (
+                    <Button
+                      h={6}
+                      size="sm"
+                      colorScheme="teal"
+                      disabled={data.pool.id === '' || data.pool.isLoading}
+                      isLoading={data.pool.isLoading}
+                      onClick={() => prepareDelegationTx()}
```
Comment (pczeglik-iohk) [inline src/ui/app/components/transactionBuilder.jsx:397]: why not `onClick={prepareDelegationTx}`?

File: src/ui/app/components/transactionBuilder.jsx (hunk 1 above)
Comment (renanvalentin) [reply src/ui/app/components/transactionBuilder.jsx]: Indeed, it is not clear. improved it here: ce24811a58c623edbe48a8a75c471e65cae295f1

File: src/api/extension/index.js [hunk 3]
```diff
@@ -136,6 +136,21 @@ export const getDelegation = async () => {
   };
 };
 
+export const getPoolMetadata = async (poolId) => {
+  const delegation = await blockfrostRequest(`/pools/${poolId}/metadata`);
```
[the remark this answers is not in the document]
Comment (pczeglik-iohk) [reply src/api/extension/index.js:140]: yes there is, but you can still import this function in different place and you can execute it without providing guard, right?

**Comment by pczeglik-iohk** [comment]
> > Why do we need typescript setup?
> 
> My original idea was to migrate a few bits to TS, but it turns out it would slightly increase the scope of this task. At least we have it in place now.

I would vote for not committing the code which is not used. It's not a huge change, so in the future if we decide to migrate this codebase to TS or start using TS for new code, it is not a big deal to configure it.

wdyt?

File: src/ui/app/pages/wallet.jsx [hunk 4]
```diff
@@ -269,22 +270,20 @@ const Wallet = () => {
...
-                      size="xs"
-                      colorScheme="whiteAlpha"
-                      rounded="lg"
-                    >
-                      Delegate
-                    </Button>
-                  )
+                  <Button
+                    onClick={() =>
+                      builderRef.current.initDelegation(
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply src/ui/app/pages/wallet.jsx:274]: Can we resolve this @pczeglik-iohk ?

File: src/ui/app/components/transactionBuilder.jsx (hunk 2 above)
Comment (rhyslbw) [reply src/ui/app/components/transactionBuilder.jsx:416]: Can we resolve this @pczeglik-iohk ?

File: src/ui/app/components/transactionBuilder.jsx [hunk 5]
```diff
@@ -268,24 +341,76 @@ const TransactionBuilder = React.forwardRef(({ onConfirm }, ref) => {
...
+                      ...s,
+                      pool: {
+                        ...s.pool,
+                        id: e.target.value,
+                        state: PoolStates.EDITING,
+                      },
+                    }));
+                  }}
+                  placeholder="Enter Pool ID"
+                  onKeyDown={(e) => {
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply src/ui/app/components/transactionBuilder.jsx:386]: Can we resolve this @pczeglik-iohk ?

File: src/ui/app/components/transactionBuilder.jsx [hunk 6]
```diff
@@ -268,24 +341,76 @@ const TransactionBuilder = React.forwardRef(({ onConfirm }, ref) => {
...
-            <Image src={IOHK} width="40px" />
-            <Box h="4" />
             <Text fontSize="sm">
-              Support the development of Nami Wallet by delegating to{' '}
+              Enter the Stake Pool ticker or ID to delegate your funds and start
+              receiving rewards. Alternatively, head to{' '}
               <Link
                 fontWeight="semibold"
-                onClick={() => window.open('https://pipool.online')}
+                onClick={() => window.open('https://pool.pm')}
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply src/ui/app/components/transactionBuilder.jsx:350]: Can we resolve this @pczeglik-iohk ?

File: src/api/extension/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rhyslbw) [reply src/api/extension/index.js:140]: @renanvalentin makes sense to guard here

**Comment by renanvalentin** [comment]
@lucas-barros @pczeglik-iohk removed the typescript commit.

**Review by lucas-barros — APPROVED** [review]

**Pull request merged**